lib/travis/api/v3/queries/log.rb

```diff
@@ -40,6 +40,11 @@ def prefix
       "jobs/#{@job.id}/log.txt"
     end
 
+    def s3_config
+      conf = config.log_options.try(:s3) || {}
+      conf.merge!(bucket_name: bucket_name)
```

Comment:
I think rkh is correct, we should use merge here instead of merge! We want to modify only the local config not the Travis.config by mistake.